### PR TITLE
PERF-154 - API tests for Notes

### PR DESCRIPTION
**Pull request by mtraneis**
Created JMX scripts for testing mod-notes POST and GET /notes
Added database scripts to the **common** folder instead of in the jmeter-supported-data so that these scripts can be reused in other tests.

### CODE REVIEW

File: common/db-scripts/notes/notes-data-load.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+DBCONF=${1}
+export TENANT=${2}
+
+export PGDATABASE=`cat $DBCONF | jq '.database' | cut -d\" -f2`
+export PGUSER=`cat $DBCONF | jq '.username' | cut -d\" -f2`
+export PGPASSWORD=`cat $DBCONF | jq '.password' | cut -d\" -f2`
+export PGHOST=`cat $DBCONF | jq '.host' | cut -d\" -f2`
+export PGPORT=`cat $DBCONF | jq '.port' | cut -d\" -f2`
+
+NOTES="notes.tsv"
```
Comment (javalkar-varun) [inline common/db-scripts/notes/notes-data-load.sh:19]: Where is `notes.tsv`? In common/db-scripts/notes/notes.json.tar.gz?

File: api-scripts/notes/mod-notes_GET_POST.jmx [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    &quot;middleName&quot;: &quot;PTF&quot;&#xd;
+  },&#xd;
+  &quot;status&quot; : &quot;ASSIGNED&quot;,&#xd;
+  &quot;links&quot;: [&#xd;
+    {&#xd;
+      &quot;id&quot; : &quot;583-2356521&quot;,&#xd;
+      &quot;type&quot;: &quot;package&quot;&#xd;
+    },&#xd;
+    {&#xd;
+      &quot;id&quot; : &quot;583-2356521-758038&quot;,&#xd;
```
Comment (javalkar-varun) [inline api-scripts/notes/mod-notes_GET_POST.jmx:274]: POST request body is hardcoded. Is it required?

File: api-scripts/notes/mod-notes_GET_POST.jmx [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+          <hashTree/>
+        </hashTree>
+        <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="mod-notes POST" enabled="true">
+          <boolProp name="HTTPSampler.postBodyRaw">true</boolProp>
+          <elementProp name="HTTPsampler.Arguments" elementType="Arguments">
+            <collectionProp name="Arguments.arguments">
+              <elementProp name="" elementType="HTTPArgument">
+                <boolProp name="HTTPArgument.always_encode">false</boolProp>
+                <stringProp name="Argument.value">{&#xd;
+  &quot;type&quot;: &quot;High Priority&quot;,&#xd;
```
Comment (javalkar-varun) [inline api-scripts/notes/mod-notes_GET_POST.jmx:253]: Are any of these field values required to be unique?

File: common/db-scripts/notes/notes-data-load.sh (hunk 1 above)
Comment (mtraneis) [reply common/db-scripts/notes/notes-data-load.sh:19]: Yes. I should make a note of that in the README

File: api-scripts/notes/mod-notes_GET_POST.jmx (hunk 2 above)
Comment (mtraneis) [reply api-scripts/notes/mod-notes_GET_POST.jmx:274]: The POST body is required to create a new record. It contains required fields and optional fields to give the record some size.

File: api-scripts/notes/mod-notes_GET_POST.jmx (hunk 3 above)
Comment (mtraneis) [reply api-scripts/notes/mod-notes_GET_POST.jmx:253]: No, the only unique field is UUID of the note, but it is auto-supplied when submitting a POST request.

File: common/db-scripts/notes/notes-data-load.sh (hunk 1 above)
Comment (mtraneis) [reply common/db-scripts/notes/notes-data-load.sh:20]: Noted where to find notes.tsv in this file instead. The README already contains instructions to unzip the file.

File: api-scripts/notes/mod-notes_GET_POST.jmx (hunk 3 above)
Comment (javalkar-varun) [reply api-scripts/notes/mod-notes_GET_POST.jmx:253]: Ok. In a real scenario, all notes will not be the same. Not sure if creating all these notes with unique content will show true performance as in a production environment. If it doesn't matter then it should be fine.

File: api-scripts/notes/mod-notes_GET_POST.jmx (hunk 2 above)
Comment (javalkar-varun) [reply api-scripts/notes/mod-notes_GET_POST.jmx:274]: Ok. In a real scenario, all notes will not be the same. Not sure if creating all these notes with unique content will show true performance as in a production environment. If it doesn't matter then it should be fine.

**Review by javalkar-varun — APPROVED** [review]

**Review by Roman-Fedynyshyn — APPROVED** [review]

**Pull request merged**